src/controllers/info.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from "express";
+import { getHostname } from "../config";
+
+export function idGet(req: Request, res: Response) {
+  res.json(buildClientIdDocument())
+}
+
+export function buildClientIdDocument(): ClientIdDocument {
+ return {
+    "@context": [
+      "https://www.w3.org/ns/solid/oidc-context.jsonld"
+    ],
+    "client_id": `${getHostname()}/info/id`,
```

Comment:
Suggest you refactor this to export this ClientID value to as an app constant (as you're gonna need it when you login (and in your test!)).